--- origsrc/util-linux-ng-2.14.1/login-utils/agetty.c	2008-05-28 18:01:02.000000000 -0500
+++ src/util-linux-ng-2.14.1/login-utils/agetty.c	2009-01-06 14:17:22.095250000 -0600
@@ -37,7 +37,7 @@
 #include "nls.h"
 #include "pathnames.h"
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
 #include <sys/param.h>
 #define USE_SYSLOG
 #endif
@@ -282,7 +282,7 @@ main(argc, argv)
 
     parse_args(argc, argv, &options);
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
 	setsid();
 #endif
 	
@@ -1111,7 +1116,7 @@ termio_final(op, tp, cp)
     tp->c_cc[VQUIT] = DEF_QUIT;			/* default quit */
     tp->c_cc[VEOF] = DEF_EOF;			/* default EOF character */
     tp->c_cc[VEOL] = DEF_EOL;
-#ifdef __linux__
+#if defined(VSWTC)
     tp->c_cc[VSWTC] = DEF_SWITCH;		/* default switch character */
 #elif defined(VSWTCH)
     tp->c_cc[VSWTCH] = DEF_SWITCH;		/* default switch character */
