--- origsrc/util-linux-2.21/term-utils/agetty.c	2012-02-28 21:59:34.971613600 -0600
+++ src/util-linux-2.21/term-utils/agetty.c	2012-02-28 21:51:00.698227900 -0600
@@ -58,6 +58,9 @@
 #    define DEFAULT_STERM  "vt102"
 #  endif
 #else
+#  ifdef __CYGWIN__
+#    define USE_SYSLOG
+#  endif
 #  ifndef DEFAULT_VCTERM
 #    define DEFAULT_VCTERM "vt100"
 #  endif
@@ -846,7 +849,9 @@ static void update_utmp(struct options *
 	ut.ut_time = t;
 	ut.ut_type = LOGIN_PROCESS;
 	ut.ut_pid = pid;
+#ifndef __CYGWIN__
 	ut.ut_session = sid;
+#endif
 
 	pututline(&ut);
 	endutent();
@@ -938,7 +943,11 @@ static void open_tty(char *tty, struct t
 			 */
 			if (vhangup())
 				log_err("/dev/%s: vhangup() failed: %m", tty);
+#ifdef TIOCNOTTY
 			ioctl(fd, TIOCNOTTY);
+#else
+			setsid();
+#endif
 		}
 
 		close(fd);
@@ -1165,7 +1174,11 @@ static void reset_vc(const struct option
 	tp->c_oflag &= ~(OLCUC | OCRNL | ONOCR | ONLRET | OFILL | \
 			    NLDLY|CRDLY|TABDLY|BSDLY|VTDLY|FFDLY);
 	tp->c_lflag |=  (ISIG | ICANON | IEXTEN | ECHO|ECHOE|ECHOK|ECHOKE);
+#ifdef ECHOPRT
 	tp->c_lflag &= ~(ECHONL|ECHOCTL|ECHOPRT | NOFLSH | TOSTOP);
+#else
+	tp->c_lflag &= ~(ECHONL|ECHOCTL| NOFLSH | TOSTOP);
+#endif
 
 	if ((op->flags & F_KEEPCFLAGS) == 0) {
 		tp->c_cflag |=  (CREAD | CS8 | HUPCL);
@@ -1559,7 +1572,7 @@ static void termio_final(struct options
 	tp->c_cc[VQUIT] = DEF_QUIT;
 	tp->c_cc[VEOF] = DEF_EOF;
 	tp->c_cc[VEOL] = DEF_EOL;
-#ifdef __linux__
+#if defined(VSWTC)
 	tp->c_cc[VSWTC] = DEF_SWITCH;
 #elif defined(VSWTCH)
 	tp->c_cc[VSWTCH] = DEF_SWITCH;
