--- origsrc/util-linux-2.21/misc-utils/cal.c	2012-01-09 08:53:10.326391300 -0600
+++ src/util-linux-2.21/misc-utils/cal.c	2012-02-28 21:12:37.476762800 -0600
@@ -297,7 +297,7 @@ main(int argc, char **argv) {
  * the locale database, which can be overridden with the
  * -s (Sunday) or -m (Monday) options.
  */
-#ifdef HAVE_DECL__NL_TIME_WEEK_1STDAY
+#if HAVE_DECL__NL_TIME_WEEK_1STDAY
 	/*
 	 * You need to use 2 locale variables to get the first day of the week.
 	 * This is needed to support first_weekday=2 and first_workday=1 for
