--- origsrc/util-linux-2.21/sys-utils/Makefile.am	2012-01-23 13:48:08.111676200 -0600
+++ src/util-linux-2.21/sys-utils/Makefile.am	2012-02-28 21:19:54.317605200 -0600
@@ -2,11 +2,10 @@ include $(top_srcdir)/config/include-Mak
 
 bin_PROGRAMS =
 sbin_PROGRAMS =
-usrbin_exec_PROGRAMS = flock ipcrm ipcs ipcmk renice setsid
-usrsbin_exec_PROGRAMS = readprofile
+usrbin_exec_PROGRAMS = flock ipcmk renice setsid
+usrsbin_exec_PROGRAMS =
 
-dist_man_MANS = flock.1 ipcrm.1 ipcs.1 ipcmk.1 renice.1 setsid.1 \
-		readprofile.8
+dist_man_MANS = flock.1 ipcmk.1 renice.1 setsid.1
 
 if LINUX
 #
@@ -15,10 +14,11 @@ if LINUX
 #
 bin_PROGRAMS += dmesg
 sbin_PROGRAMS += ctrlaltdel fsfreeze fstrim
-usrbin_exec_PROGRAMS += cytune setarch
-usrsbin_exec_PROGRAMS += ldattach tunelp rtcwake
+usrbin_exec_PROGRAMS += cytune ipcrm ipcs setarch
+usrsbin_exec_PROGRAMS += ldattach tunelp readprofile rtcwake
 
 dist_man_MANS += dmesg.1 ctrlaltdel.8 cytune.8 setarch.8 \
+		ipcrm.1 ipcs.1 readprofile.8 \
 		ldattach.8 tunelp.8 rtcwake.8 fsfreeze.8 fstrim.8
 endif # LINUX
 
@@ -168,6 +168,7 @@ bin_PROGRAMS += arch
 dist_man_MANS += arch.1
 endif
 
+if LINUX
 SETARCH_LINKS = linux32 linux64
 
 if ARCH_S390
@@ -201,6 +202,7 @@ CLEANFILES = $(man_MANS)
 
 $(SETARCH_MAN_LINKS):
 	$(AM_V_GEN)echo ".so man8/setarch.8" > $@
+endif # LINUX
 
 install-exec-hook:
 if BUILD_SWAPON
